bin/olly.ml

```diff
@@ -7,7 +7,18 @@ open Cmdliner
 type trace_format = Json | Fuchsia
 
 let total_gc_time = Atomic.make 0
-let start_time = Unix.gettimeofday ()
+let start_time = ref 0.0
+
+let lifecycle _domain_id _ts lifecycle_event data =
+  match lifecycle_event with
+  | Runtime_events.EV_RING_START ->
+      begin
+          assert(match data with
+          | Some _ -> true
+          | None -> false);
+          start_time := Unix.gettimeofday ()
+      end
+  | _ -> ()
```

Comment:
Why isn't end time calculated using `EV_RING_END`? I guess if the program is ctrl+c'ed then you'll not have a `EV_RING_END`?

Reply:
I guess we don't print any stats when the program is Ctrl+c'ed. Probably it's better to track the end time with `EV_RING_END`. Done.